### ISSUE TITLE
[R] Refer to the Arrow C++ compute engine as **Acero**

### ISSUE THREAD

**Issue by ianmcook** [apache/arrow-cookbook#212]
Once [ARROW-16632](https://issues.apache.org/jira/browse/ARROW-16632) and [ARROW-16626](https://issues.apache.org/jira/browse/ARROW-16626) are complete, we should update some of the text in the R cookbook to refer to the Arrow C++ compute engine using its new name, **Acero**. We should try to be quite explicit about the fact that the {arrow} R package converts {dplyr} code into an Acero ExecPlan.

**Comment by ianmcook** [comment]
> can you expand upon where
> 
> > We should try to be quite explicit about the fact that the {arrow} R package converts {dplyr} code into an Acero ExecPlan.
> 
> would fit, in terms of where in the cookbook it'd be relevant to mention this?

Yes, I think it would be valuable to add some content about this to the [Manipulating Data - Tables](https://arrow.apache.org/cookbook/r/manipulating-data---tables.html) R cookbook entry. It would be great if that cookbook entry could explain what happens under the hood when the user calls dplyr verbs on an Arrow table/recordbatch/dataset and then calls `collect()` or `compute()`: an Acero ExecPlan is constructed and executed.

**Comment by thisisnic** [comment]
OK, make sense, we could mention that briefly in the intro and then link to some docs where it's explained in more detail.
